tests/geobase/checks_FairGeoBasicShape.h

```diff
@@ -0,0 +1,161 @@
+                REQUIRE_THAT(test->Y(), Catch::Matchers::WithinRel(parameters.at(1).at(i), 0.001f));
+                REQUIRE_THAT(test->Z(), Catch::Matchers::WithinRel(parameters.at(2).at(i), 0.001f));
+            } else {
+                REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+                REQUIRE_THAT(test->Y(), Catch::Matchers::WithinRel(parameters.at(1).at(i), 0.001f));
+            }
+        } else if ("TORUS" == shapeName) {
+            REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+        } else if ("SPHE" == shapeName) {
+            REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+            REQUIRE_THAT(test->Y(), Catch::Matchers::WithinRel(parameters.at(1).at(i), 0.001f));
+        } else if ("PCON" == shapeName) {
+            if (0 == i) {
+                REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+            } else if (1 == i) {
+                REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+                REQUIRE_THAT(test->Y(), Catch::Matchers::WithinRel(parameters.at(1).at(i), 0.001f));
+            } else {
+                REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+                REQUIRE_THAT(test->Y(), Catch::Matchers::WithinRel(parameters.at(1).at(i), 0.001f));
+                REQUIRE_THAT(test->Z(), Catch::Matchers::WithinRel(parameters.at(2).at(i), 0.001f));
+            }
+        } else if ("PGON" == shapeName) {
+            if (0 == i) {
+                REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+            } else {
+                REQUIRE_THAT(test->X(), Catch::Matchers::WithinRel(parameters.at(0).at(i), 0.001f));
+                REQUIRE_THAT(test->Y(), Catch::Matchers::WithinRel(parameters.at(1).at(i), 0.001f));
+                REQUIRE_THAT(test->Z(), Catch::Matchers::WithinRel(parameters.at(2).at(i), 0.001f));
+            }
+        }
+    }
+
+    // Use the print function to generate the screen output
+    // Currently I don't know how to test this
+    // Probably not very important since the functionality isn't
+    // really used
+    shape.printPoints(&volume);
+    shape.printParam();
+
+    // Write the parmaters to an output file
+    std::string testFileName = fairroot::tests::generateTempFileName();
+    std::fstream testFile{};
+    testFile.open(testFileName, std::ios::out);
+    shape.writePoints(&testFile, &volume);
+    testFile.close();
+
+    // Check if the two files are identical
+    REQUIRE(fairroot::tests::compareFiles(fileName, testFileName));
+
+    // Remove the temporary files
+    //    std::cout << shapeName << ": " << fileName.c_str() << "\n";
+    std::remove(fileName.c_str());
+    std::remove(testFileName.c_str());
```

Comment:
```suggestion
    CHECK(std::remove(fileName.c_str()) == 0);
    CHECK(std::remove(testFileName.c_str()) == 0);
```

`#include <cstdio>`?

Reply:
done